### ISSUE TITLE
missing education codes

### ISSUE THREAD

**Issue by ylelkes** [kuriwaki/ccesMRPprep#12]
Hi there,
Any idea why acscodes_age_sex_educ only returns rows for "HS or Less" and "4-Year"?

acs_tab <- get_acs_cces(
  varlist = acscodes_sex_educ_race,
  varlab_df = acscodes_df,
  year = 2021,dataset = "acs5"
)
<img width="708" alt="Screenshot 2023-05-08 at 3 02 13 PM" src="https://user-images.githubusercontent.com/1694330/236909807-5ee38c81-d301-4cfe-bd36-777cdc5f733f.png">

thanks!

**Comment by kuriwaki** [comment]
The recodings of the education to factors were incorrect. It should have been that for this partition,

table of codes ending in: associated education level
* `_006`: `Bachelor's degree or higher`
* `_011`: `Bachelor's degree or higher`
* `_004`: `High school graduate (includes equivalency)  `
* `_009`: `High school graduate (includes equivalency) `
* `_005`: `Some college or associate's degree`
* `_010`: `Some college or associate's degree`

Below is a demo without the `get_acs_cces` wrapper.

I will try to fix it soon in `dev`, but I might need to do be creative since the way education is grouped here is different from the other partition `acscodes_age_sex_educ` that we were using in the paper. Apologies.

Variables 

``` r
library(ccesMRPprep)
library(tidycensus)
library(dplyr)

acs5_vars <- load_variables("acs5", year = 2021)

acs5_vars |> 
  filter(name %in% acscodes_sex_educ_race)
#> # A tibble: 48 × 4
#>    name        label                                           concept geography
#>    <chr>       <chr>                                           <chr>   <chr>    
#>  1 C15002B_004 Estimate!!Total:!!Male:!!High school graduate … SEX BY… tract    
#>  2 C15002B_005 Estimate!!Total:!!Male:!!Some college or assoc… SEX BY… tract    
#>  3 C15002B_006 Estimate!!Total:!!Male:!!Bachelor's degree or … SEX BY… tract    
#>  4 C15002B_009 Estimate!!Total:!!Female:!!High school graduat… SEX BY… tract    
#>  5 C15002B_010 Estimate!!Total:!!Female:!!Some college or ass… SEX BY… tract    
#>  6 C15002B_011 Estimate!!Total:!!Female:!!Bachelor's degree o… SEX BY… tract    
#>  7 C15002C_004 Estimate!!Total:!!Male:!!High school graduate … SEX BY… tract    
#>  8 C15002C_005 Estimate!!Total:!!Male:!!Some college or assoc… SEX BY… tract    
#>  9 C15002C_006 Estimate!!Total:!!Male:!!Bachelor's degree or … SEX BY… tract    
#> 10 C15002C_009 Estimate!!Total:!!Female:!!High school graduat… SEX BY… tract    
#> # ℹ 38 more rows
```

<sup>Created on 2023-05-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by ylelkes** [comment]
Thank you!

From: Shiro Kuriwaki ***@***.***>
Date: Monday, May 8, 2023 at 10:33 PM
To: kuriwaki/ccesMRPprep ***@***.***>
Cc: ylelkes ***@***.***>, Author ***@***.***>
Subject: Re: [kuriwaki/ccesMRPprep] missing education codes (Issue #12)

The recodings of the education to factors were incorrect. It should have been that for this partition,

table of codes ending in: associated education level

  *   _006: Bachelor's degree or higher
  *   _011: Bachelor's degree or higher
  *   _004: High school graduate (includes equivalency)
  *   _009: High school graduate (includes equivalency)
  *   _005: Some college or associate's degree
  *   _010: Some college or associate's degree

Below is a demo without the get_acs_cces wrapper.

I will try to fix it soon in dev, but I might need to do be creative since the way education is grouped here is different from the other partition acscodes_age_sex_educ that we were using in the paper. Apologies.

Variables

library(ccesMRPprep)

library(tidycensus)

library(dplyr)



acs5_vars <- load_variables("acs5", year = 2021)



acs5_vars |>

  filter(name %in% acscodes_sex_educ_race)

#> # A tibble: 48 × 4

#>    name        label                                           concept geography

#>    <chr>       <chr>                                           <chr>   <chr>

#>  1 C15002B_004 Estimate!!Total:!!Male:!!High school graduate … SEX BY… tract

#>  2 C15002B_005 Estimate!!Total:!!Male:!!Some college or assoc… SEX BY… tract

#>  3 C15002B_006 Estimate!!Total:!!Male:!!Bachelor's degree or … SEX BY… tract

#>  4 C15002B_009 Estimate!!Total:!!Female:!!High school graduat… SEX BY… tract

#>  5 C15002B_010 Estimate!!Total:!!Female:!!Some college or ass… SEX BY… tract

#>  6 C15002B_011 Estimate!!Total:!!Female:!!Bachelor's degree o… SEX BY… tract

#>  7 C15002C_004 Estimate!!Total:!!Male:!!High school graduate … SEX BY… tract

#>  8 C15002C_005 Estimate!!Total:!!Male:!!Some college or assoc… SEX BY… tract

#>  9 C15002C_006 Estimate!!Total:!!Male:!!Bachelor's degree or … SEX BY… tract

#> 10 C15002C_009 Estimate!!Total:!!Female:!!High school graduat… SEX BY… tract

#> # ℹ 38 more rows

Created on 2023-05-08 with reprex v2.0.2<https://reprex.tidyverse.org>

**Comment by kuriwaki** [comment]
I started some edits in https://github.com/kuriwaki/ccesMRPprep/tree/fix_iss12 which I will update and merge. 

@ylelkes thanks for the suggested code. It seems like in this case, education is the three way coding where BA and post-grads is one level. Which means that the CES survey data will need to be recoded to fit with that coding.  This goes to what I am contemplating: whether to try and distinguish between 4-way education and `educ_3` which is this three-way coding in the output of `get_acs_cces`, OR whether to try and coalesce to one (so the user will need to know how to recode education in the CES side).  

The edits in the branch currently does the latter

**Comment by kuriwaki** [comment]
I changed plans and will do the former -- formally distinguish between `educ` (4-way) and `educ_3` (3-way). This way there is no ambiguity between the two types when obtaining data. I added `educ_3` as a standard demographic in ccc_std_demographics(). 

``` r
library(ccesMRPprep)
library(dplyr)
packageVersion("ccesMRPprep")
#> [1] '0.1.11.9999'

acs_tab <- get_acs_cces(
  varlab_df = acscodes_df,
  varlist = acscodes_sex_educ_race,
  year = 2021,
  dataset = "acs5"
)
#> Getting data from the 2017-2021 5-year ACS
# no educ, only educ_3
acs_tab
#> # A tibble: 28,096 × 9
#>    acscode      year cd    gender female educ_3            race  count count_moe
#>    <chr>       <dbl> <chr> <fct>   <int> <fct>             <fct> <dbl>     <dbl>
#>  1 C15002B_003  2021 AL-01 Male        0 HS or Less        Black  9804       713
#>  2 C15002B_004  2021 AL-01 Male        0 HS or Less        Black 25034      1256
#>  3 C15002B_005  2021 AL-01 Male        0 Some College      Black 14180       995
#>  4 C15002B_006  2021 AL-01 Male        0 4-Year or Post-G… Black  6767       804
#>  5 C15002B_008  2021 AL-01 Female      1 HS or Less        Black  8304       734
#>  6 C15002B_009  2021 AL-01 Female      1 HS or Less        Black 23756      1237
#>  7 C15002B_010  2021 AL-01 Female      1 Some College      Black 23514       945
#>  8 C15002B_011  2021 AL-01 Female      1 4-Year or Post-G… Black 13255       840
#>  9 C15002C_003  2021 AL-01 Male        0 HS or Less        Nati…   520       200
#> 10 C15002C_004  2021 AL-01 Male        0 HS or Less        Nati…   624       168
#> # ℹ 28,086 more rows
count(acs_tab, educ_3)
#> # A tibble: 3 × 2
#>   educ_3                  n
#>   <fct>               <int>
#> 1 HS or Less          14048
#> 2 Some College         7024
#> 3 4-Year or Post-Grad  7024


# This one gets you four-way education
get_acs_cces(
  varlab_df = acscodes_df,
  varlist = acscodes_age_sex_educ,
  year = 2021,
  dataset = "acs5"
)
#> Getting data from the 2017-2021 5-year ACS
#> # A tibble: 30,730 × 9
#>    acscode     year cd    gender female educ         age         count count_moe
#>    <chr>      <dbl> <chr> <fct>   <int> <fct>        <fct>       <dbl>     <dbl>
#>  1 B15001_004  2021 AL-01 Male        0 HS or Less   18 to 24 y…   329       146
#>  2 B15001_005  2021 AL-01 Male        0 HS or Less   18 to 24 y…  4670       591
#>  3 B15001_006  2021 AL-01 Male        0 HS or Less   18 to 24 y… 12135       801
#>  4 B15001_007  2021 AL-01 Male        0 Some College 18 to 24 y…  9791       841
#>  5 B15001_008  2021 AL-01 Male        0 Some College 18 to 24 y…  1507       361
#>  6 B15001_009  2021 AL-01 Male        0 4-Year       18 to 24 y…  1379       324
#>  7 B15001_010  2021 AL-01 Male        0 Post-Grad    18 to 24 y…    65        44
#>  8 B15001_012  2021 AL-01 Male        0 HS or Less   25 to 34 y…  1009       281
#>  9 B15001_013  2021 AL-01 Male        0 HS or Less   25 to 34 y…  4920       742
#> 10 B15001_014  2021 AL-01 Male        0 HS or Less   25 to 34 y… 15597      1116
#> # ℹ 30,720 more rows

# ccc_std_demographics provides BOTH educ and educ_3
ccc_std_demographics(ccc_samp) |> 
  count(educ, educ_3)
#> age variable modified to bins. Original age variable is now in age_orig.
#> # A tibble: 4 × 3
#>   educ             educ_3                      n
#>   <dbl+lbl>        <dbl+lbl>               <int>
#> 1 1 [HS or Less]   1 [HS or Less]            293
#> 2 2 [Some College] 2 [Some College]          355
#> 3 3 [4-Year]       3 [4-Year or Post-Grad]   244
#> 4 4 [Post-Grad]    3 [4-Year or Post-Grad]   108
```

<sup>Created on 2023-06-19 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by kuriwaki** [comment]
Closing this as it was addressed in #13 last year. Thank you @ylelkes for finding the mistake.
